src/library/Uno.Material/Controls/Chips/ChipGroup.cs

```diff
@@ -0,0 +1,371 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Uno.Extensions.Specialized;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Markup;
+
+namespace Uno.Material.Controls
+{
+	public partial class ChipGroup : ItemsControl
+	{
+		public event ChipItemEventHandler ItemClick;
+		public event ChipItemEventHandler ItemChecked;
+		public event ChipItemEventHandler ItemUnchecked;
+
+		#region DependencyProperty: SelectedItem
+
+		public static DependencyProperty SelectedItemProperty { get; } = DependencyProperty.Register(
+			nameof(SelectedItem),
+			typeof(object),
+			typeof(ChipGroup),
+			new PropertyMetadata(default, (s, e) => (s as ChipGroup)?.OnSelectedItemChanged(e)));
+
+		public object SelectedItem
+		{
+			get => (object)GetValue(SelectedItemProperty);
+			set => SetValue(SelectedItemProperty, value);
+		}
+
+		#endregion
+		#region DependencyProperty: SelectedItems
+
+		public static DependencyProperty SelectedItemsProperty { get; } = DependencyProperty.Register(
+			nameof(SelectedItems),
+			typeof(IList),
+			typeof(ChipGroup),
+			new PropertyMetadata(default, (s, e) => (s as ChipGroup)?.OnSelectedItemsChanged(e)));
+
+		public IList SelectedItems
+		{
+			get => (IList)GetValue(SelectedItemsProperty);
+			set => SetValue(SelectedItemsProperty, value);
+		}
+
+		#endregion
+		#region DependencyProperty: SelectionMemberPath
+
+		public static DependencyProperty SelectionMemberPathProperty { get; } = DependencyProperty.Register(
+			nameof(SelectionMemberPath),
+			typeof(string),
+			typeof(ChipGroup),
+			new PropertyMetadata(default, (s, e) => (s as ChipGroup)?.OnSelectionMemberPathChanged(e)));
+
+		/// <summary>
+		/// Gets or sets the name or path of the property that indicates selection state for each data item.
+		/// </summary>
+		public string SelectionMemberPath
+		{
+			get => (string)GetValue(SelectionMemberPathProperty);
+			set => SetValue(SelectionMemberPathProperty, value);
+		}
+
+		#endregion
+		#region DependencyProperty: SelectionMode = ChipSelectionMode.Single
+
+		public static DependencyProperty SelectionModeProperty { get; } = DependencyProperty.Register(
+			nameof(SelectionMode),
+			typeof(ChipSelectionMode),
+			typeof(ChipGroup),
+			new PropertyMetadata(ChipSelectionMode.Single, (s, e) => (s as ChipGroup)?.OnSelectionModeChanged(e)));
+
+		public ChipSelectionMode SelectionMode
+		{
+			get => (ChipSelectionMode)GetValue(SelectionModeProperty);
+			set => SetValue(SelectionModeProperty, value);
+		}
+
+		#endregion
+
+		private bool _isSynchronizingSelection = false;
+		private bool _isUpdatingSelection = false;
+
+		public ChipGroup()
+		{
+#if WINDOWS_UWP
```

Comment:
it's not needed in Uno platforms? why? (possibly add comment to explain)

Reply:
Uno exposes OnItemsSourceChanged, but the equivalent doesnt exist on uwp.
Actually, it should be cleaner to use just `RegisterPropertyChangedCallback` and forget `ItemsControl::OnItemsSourceChanged`.